src/models/v3/asyncapi.ts

```diff
@@ -1,7 +1,164 @@
+import { Schemas } from './schemas';
 
-export class AsyncAPIDocument extends BaseModel {
+import { extensions } from './mixins';
+import { tilde } from '../../utils';
+import { SchemaTypesToIterate, traverseAsyncApiDocument } from '../../iterator';
+
+import type { AsyncAPIDocumentInterface } from '../asyncapi';
+import type { InfoInterface } from '../info';
+import type { ServersInterface } from '../servers';
+import type { ChannelsInterface } from '../channels';
+import type { OperationsInterface } from '../operations';
+import type { MessagesInterface } from '../messages';
+import type { MessageInterface } from '../message';
+import type { ComponentsInterface } from '../components';
+import type { SecuritySchemesInterface } from '../security-schemes';
+import type { ExtensionsInterface } from '../extensions';
+import type { SchemaInterface } from 'models/schema';
+import type { SchemasInterface } from 'models/schemas';
+import type { v3 } from '../../spec-types';
+import { OperationInterface } from 'models/operation';
+import { ChannelInterface } from 'models/channel';
+import { ServerInterface } from 'models/server';
+
+export class AsyncAPIDocument extends BaseModel<v3.AsyncAPIObject> implements AsyncAPIDocumentInterface {
   version(): string {
     return this._json.asyncapi;
   }
+
+  defaultContentType(): string | undefined {
+    return this._json.defaultContentType;
+  }
+
+  hasDefaultContentType(): boolean {
+    return !!this._json.defaultContentType;
+  }
+
+  info(): InfoInterface {
+    return this.createModel(Info, this._json.info, { pointer: '/info' });
+  }
+
+  servers(): ServersInterface {
+    return new Servers(
+      Object.entries(this._json.servers || {}).map(([serverName, server]) => 
+        this.createModel(Server, server as v3.ServerObject, { id: serverName, pointer: `/servers/${tilde(serverName)}` })
+      )
+    );
+  }
+
+  channels(): ChannelsInterface {
+    return new Channels(
+      Object.entries(this._json.channels || {}).map(([channelId, channel]) =>
+        this.createModel(Channel, channel as v3.ChannelObject, { id: channelId, pointer: `/channels/${tilde(channelId)}` })
+      )
+    );
+  }
+
+  operations(): OperationsInterface {    
+    return new Operations(
+      Object.entries(this._json.operations || {}).map(([operationId, operation]) =>
+        this.createModel(Operation, operation as v3.OperationObject, { id: operationId, pointer: `/operations/${tilde(operationId)}` })
+      )
+    );
+  }
+
+  messages(): MessagesInterface {
+    const messages: MessageInterface[] = [];
+    const messagesData: any[] = [];
+    this.channels().forEach(channel => {
+      channel.messages().forEach(message => {
+        const messageData = message.json();
+        if (!messagesData.includes(messageData)) {
+          messagesData.push(messageData);
+          messages.push(message);
+        }
+      });
+    });
+    return new Messages(messages);
+  }
+
+  schemas() {
+    return null as any;
```

Comment:
This seems wrong

Reply:
Fixed in https://github.com/asyncapi/parser-js/pull/836